ext/lmdb-safe/lmdb-typed.cc

```diff
@@ -11,4 +13,18 @@ unsigned int MDBGetMaxID(MDBRWTransaction& txn, MDBDbi& dbi)
   return maxid;
 }
 
+unsigned int MDBGetRandomID(MDBRWTransaction& txn, MDBDbi& dbi)
+{
+  auto cursor = txn->getRWCursor(dbi);
+  unsigned int id;
+  for(int attempts=0; attempts<20; attempts++) {
+    MDBOutVal key, content;
+    id = dns_random(std::numeric_limits<decltype(id)>::max()-1) + 1;
```

Comment:
This generates number in the range [1..MAX_UINT-1]. Is that intended? If so, some explanation why 0 and MAX_UINT are left out would be nice.

Reply:
I believe it generates in [1..MAX_UINT]. Please confirm :) I'll add a comment with the why.

Reply:
`dns_random(upper_bound)` generates number in the range `[0.. upperbound-1]` so `dns_random(MAX_UINT-1])` generates in the range `[0..MAX_UINT-2]`, Add 1 and you get `[1..MAX_UINT-1]`.

Reply:
0 is used to indicate "we haven't got an ID yet, so please give us a new one.". And doesn't the +1 at the end make sure it does go up to MAX_UINT?

Reply:
thanks, pushed a fix and some extra comment text